Add RC1 release notes to 10.2.0 release notes page (#3298)
akka/akka-http: docs/src/main/paradox/release-notes/10.2.x.md

# 10.2.x Release Notes

Among other things, 10.2.0-RC1 contains:

> * Improved API's and documentation to improve the experience working with Akka 2.6 and the new Actor API's. Akka HTTP 10.2.0 will keep supporting Akka 2.5.

Reviewer: ```suggestion
* APIs and documentation now provide a seemless experience with Akka 2.6 and the new Actor APIs. Akka HTTP 10.2.x will still be supporting Akka 2.5 to ease incremental updating.
```

* Silence outgoing request stream error [#2905](https://github.com/akka/akka-http/pull/2905)
* Add SameSite attribute to Cookies [#2928](https://github.com/akka/akka-http/pull/2928)
* Only catch NonFatal Exceptions [#2853](https://github.com/akka/akka-http/pull/2853)
* Add coordinated shutdown support [#3142](https://github.com/akka/akka-http/pull/3142)
* Percent-encode illegal chars when creating URI query [#3003](https://github.com/akka/akka-http/pull/3003)
* Add the remote address in parse errors when possible [#2899](https://github.com/akka/akka-http/pull/2899)
* Remove `UseHttp2` [#2896](https://github.com/akka/akka-http/pull/2896)
* Hide body and headers by default in `HttpRequest#toString` and `HttpResponse#toString` [#2560](https://github.com/akka/akka-http/pull/2560)
* Fix headers javadsl scaladoc [#2932](https://github.com/akka/akka-http/pull/2932)

> * Allow client setting overrides for target hosts in config [#2574](https://github.com/akka/akka-http/pull/2574)

Reviewer: Maybe put that one into the headline features. It had been breeding for long and is a really useful feature.

* Better support for the new Actors API [#3036](https://github.com/akka/akka-http/pull/3036)
* Parse empty query `?` to `Query.empty` [#3042](https://github.com/akka/akka-http/pull/3042)
* Make sure to cancel response entity on failure [#3046](https://github.com/akka/akka-http/pull/3046)
* Add `akka.http.server.remote-address-attribute` [#2924](https://github.com/akka/akka-http/pull/2924)
* Make `transparent-head-requests` opt-in [#3063](https://github.com/akka/akka-http/pull/3063)

> * Allow multiple Host headers in responses [#3158](https://github.com/akka/akka-http/pull/3158)

Reviewer: That sounds like bug ;)

```suggestion
* Continue more gracefully when encountering multiple Host headers in a response [#3158](https://github.com/akka/akka-http/pull/3158)
```

* Allow customizing parsing errors [#3049](https://github.com/akka/akka-http/pull/3049)

> * Don't extend from Char => Boolean for CharPredicate [#3107](https://github.com/akka/akka-http/pull/3107)

Reviewer: Ultimately, we should point out the performance improvements as such (and not as its technical details) but can wait for the final announcement.

* Add server-side streamCancellationDelay to mitigate cancellation race conditions [#2116](https://github.com/akka/akka-http/pull/2116)
* Improve error message about unconsumed entity [#3109](https://github.com/akka/akka-http/pull/3109)
* Header rendering with less indirection [#3106](https://github.com/akka/akka-http/pull/3106)
* Remove deprecated methods taking implicit materializers [#3119](https://github.com/akka/akka-http/pull/3119)
* Simplify `superPool` and `clientFlow` [#3156](https://github.com/akka/akka-http/pull/3156)
* Deprecate Remote-Address header [#3174](https://github.com/akka/akka-http/pull/3174)

> * move `max-content-length` definition from top to server/client [#3098](https://github.com/akka/akka-http/pull/3098)

Reviewer: ```suggestion
* Move `max-content-length` definition from top to server/client [#3098](https://github.com/akka/akka-http/pull/3098)
```

* Update to Scalatest 3.1.0 [#2851](https://github.com/akka/akka-http/pull/2851)
* Handle test exceptions separately from regular exceptions [#2949](https://github.com/akka/akka-http/pull/2949)
* Simplify implicits for RouteTest [#3060](https://github.com/akka/akka-http/pull/3060)
* Fix more Scala 2.13 warnings [#3130](https://github.com/akka/akka-http/pull/3130)

#### docs

> * use new ActorSystem API [#3242](https://github.com/akka/akka-http/pull/3242)

Reviewer: ```suggestion
* Use typed ActorSystem in examples [#3242](https://github.com/akka/akka-http/pull/3242)
```

* Fix project-info links to API docs [#2857](https://github.com/akka/akka-http/pull/2857)
* Drop Scala 2.11 [#2589](https://github.com/akka/akka-http/pull/2589)

> * build: build with Scala 2.13 by default [#3126](https://github.com/akka/akka-http/pull/3126)

Reviewer: ```suggestion
* Build with Scala 2.13 by default [#3126](https://github.com/akka/akka-http/pull/3126)
```
